src/elements/grid/hooks/useGridExpandedRows.ts

```diff
@@ -60,6 +62,16 @@ export default function useGridExpandedRows<T>(
     }
   }
 
+  useEffect(() => {
+    if (!expandRowsUponSelection) return
+    const rowsToExpand = rows.reduce((accum, row) => {
+      if (!row.isSelected) return accum
+      accum[row.key] = true
+      return accum
+    }, {})
+    setExpandedRowsById(rowsToExpand)
```

Comment:
this could overwrite rows that are already expanded through other means -- if we want to handle row expansions through both selection and through doing other things, then this may need to be changed. If we are only handling expansion through selected vs non-selected then this is ok

Reply:
I was thinking that it's an either or situation. You can only choose one way to expand the rows: through an action button or through the row selection. I don't see a situation where you would enable both